account_payment_cc/account_voucher.py

```diff
@@ -0,0 +1,236 @@
+﻿# -*- coding: utf-8 -*-
+
+from openerp.osv import orm
+
+
+class account_voucher(orm.Model):
+
+    _inherit = 'account.voucher'
+
+    def action_move_line_create(self, cr, uid, ids, context=None):
+        base_func = super(account_voucher, self).action_move_line_create
+        res = base_func(cr, uid, ids, context)
+        move_line_pool = self.pool.get('account.move.line')
+        move_pool = self.pool.get('account.move')
+
+        for voucher in self.browse(cr, uid, ids):
+            if not voucher.journal_id.support_creditcard_transactions:
+                break
+
+            company_currency = self._get_company_currency(cr, uid, voucher.id,
+                                                          context)
+            current_currency = self._get_current_currency(cr, uid, voucher.id,
+                                                          context)
+
+            partner_id = voucher.journal_id.partner_id
+
+            account_payable = partner_id.property_account_payable
+            account_receivable = partner_id.property_account_receivable
+            account = voucher.journal_id.default_credit_account_id
+
+            if voucher.type in ('receipt', 'sale'):
+                account = voucher.journal_id.default_debit_account_id
```

Comment:
There is an inconsitency with the naming of the browse record variables. Some of them end with `_id` others not.
Try to stick to one naming convention.